dlt/common/storages/fsspec_filesystem.py

```diff
@@ -302,10 +312,16 @@ def glob_files(
         # make that absolute path on a file://
         if bucket_url_parsed.scheme == "file" and not file.startswith("/"):
             file = f"/{file}"
-        file_name = posixpath.relpath(file, bucket_url_no_schema)
-        file_url = bucket_url_parsed._replace(
-            path=posixpath.join(bucket_url_parsed.path, file_name)
-        ).geturl()
+
+        if is_unc_path:
+            file_name = file
```

Comment:
name must contain only name (final path component)

Reply:
good point! although our docs mention that file_name can be a relative path, it is very misleading. so let's fix it:
1. rename `file_name` to `relative_path` and add `file_name` with just the name part
2. update all the tests
3. update documentation: https://dlthub.com/docs/dlt-ecosystem/verified-sources/filesystem#fileitem-representation

luckily we do not use `file_name` in verified sources